src/validation.cpp

```diff
@@ -2991,6 +2992,10 @@ bool CChainState::ConnectBlock(const CBlock &block,
     auto isEvmEnabledForBlock = blockCtx.GetEVMEnabledForBlock();
     auto &evmTemplate = blockCtx.GetEVMTemplate();
 
+    std::vector<CEvmTxMessage> evmTxMsgs;
+    std::vector<std::vector<CEvmTxMessage>> evmTxMsgsPools;
+    TaskGroup evmEccPreCacheTaskPool;
```

Comment:
Could we define this in ConnectTip so we can call MarkCancelAndWaitForCompletion there instead of having to call it on every failure or success in ConnectBlock.